aarch64-esr-web/src/lib.rs

```diff
@@ -81,8 +96,12 @@ fn show_decoded(esr: u64, decoded: Result<Vec<FieldInfo>, DecodeError>) -> Resul
     // ESR in hexadecimal
     let row = document.create_element("tr")?;
     row.set_attribute("class", "value")?;
-    let esr_hex = format!("{:016x}", esr);
-    for digit in esr_hex.chars() {
+    let esr_hex  = format!("{:016x}", esr);
+    //loop over number of chars of interset only
+    let nr_nib = (nr_bit+3)/4;
+    let lp_st = 16-nr_nib;
+    for i in lp_st..esr_hex.chars().count() {
+        let digit = esr_hex.chars().nth(i).unwrap();
```

Comment:
I think you could make this a bit shorter and clearer by instead doing something like:

```rust
for digit in esr_hex.chars().skip(16 - nr_nib) {
```